platform/api.templates/src/org/netbeans/api/templates/CreateFromTemplateHandler.java

```diff
@@ -59,4 +64,171 @@
+            start = endPos + 1;
+            pos = start;
+        }
+        if (sb == null) {
+            return expression;
+        }
+        if (start < expression.length()) {
+            sb.append(expression.substring(start));
+        }
+        
+        return mapParameters(sb.toString(), parameters);
+    }
+    
+    private static final String PROP_TEMPLATE = "template"; // NOI18N
+
+    /**
+     * Prefix for template-controlling attributes. Such attributes will not be copied
+     * during template instantiation.
+     */
+    private static final String ATTR_TEMPLATE_PREFIX = "template.";
+    
+    /**
+     * Copies template attributes over to the created file. Copies over from the source (template) to the target (usually
+     * the produced file), except attributes that parametrize template creation. By default, attributes that are prefixed
+     * with {@code "template."} are not copied. As the operation is not atomic, the method should be called in a 
+     * {@link org.openide.filesystems.FileSystem#runAtomicAction(org.openide.filesystems.FileSystem.AtomicAction)}.
+     * The {@code "template"} attribute is never copied.
+     * 
+     * @param h the Handler responsible for file creation, usually the calling handler.
+     * @param from the original file
+     * @param to the target file
+     * @throws IOException if an I/O error occurs.
+     */
+    public static void copyAttributesFromTemplate(CreateFromTemplateHandler h, FileObject from, FileObject to) throws IOException {
+        // copy attributes; some attributes are filtered by FileSystems API already
+        FileUtil.copyAttributes(from, to);
+        // need to filter additional template-related attributes:
+        for (String a : Collections.list(to.getAttributes())) {
+            if ("javax.script.ScriptEngine".equals(a) // NOI18N 
+                || a.startsWith(ATTR_TEMPLATE_PREFIX)) {
+                to.setAttribute(a, null);
+            }
+        }
+        // hack to overcome package-private modifier in setTemplate(fo, boolean)
+        to.setAttribute(PROP_TEMPLATE, null);
+    }
+    
+    
+    /**
+     * Copies the files or folders contained in the specified folder to the target. The original descriptor can be passed in,
+     * since it has Lookup, parameters etc set up. They will be passed on to individual created files. The `contentsParent` may
+     * be {@code null}, in which case the {@link CreateDescriptor#getTemplate()} folder will be used.
+     * @param origDescriptor original descriptor to get parameters / lookup from
+     * @param contentsParent template folder whose contents should be copied
+     * @param target target folder
+     * @return created files
+     * @throws IOException 
+     * @since 1.23
+     */
+    protected static List<FileObject> defaultCopyContents(CreateDescriptor origDescriptor, FileObject contentsParent, FileObject target) throws IOException {
```

Comment:
Strange method. `protected static` is a strange combination of modifiers - if the class is subclassable, there is nothing "protected" in this method. If the class isn't subclassable, making a method `protected` makes no sense.

Reply:
Yes, you can always do a trampoline from some subclass and made it public to whatever code you like. It was meant as a helper primarily for CreateFromTemplateHandler imlementations (with limited use elsewhere), but can be `public` as well.